starter-code/src/QueueDataStructure.js

```diff
@@ -1,2 +1,30 @@
+// Queue Data Structure contructor function
 function QueueDataStructure () {
+    this.queueControl = [];
+    this.MAX_SIZE = 8;
 }
+
+QueueDataStructure.prototype.isEmpty = function() {
+    return this.queueControl == false ? true : false;
+};
+
+QueueDataStructure.prototype.canEnqueue = function() {
+    return this.queueControl.length < this.MAX_SIZE ? true : false;
+};
+
+QueueDataStructure.prototype.enqueue = function(element) {
+    if (this.queueControl.length < this.MAX_SIZE) {
+        this.queueControl.unshift(element);
+        return this.queueControl;
+    } else {
+        return 'Queue Overflow';
+    }
+};
+
+QueueDataStructure.prototype.dequeue = function() {
+    if (this.queueControl.length > 0) {
```

Comment:
Otra forma de poner esto sería la siguiente:

`if (this.queueControl.length > 0)  return this.queueControl.pop();
else return 'Queue Underflow';`

Si solo tienen una línea, se puede poner sin {}, va a gustos, pero te lo comento por si no lo conocías :)